src/patterns/strategy/interfaces.ts

```diff
@@ -0,0 +1,17 @@
+export interface IBaseDiscountStrategy {
```

Comment:
I am not sure about naming, using "base" might be considered as strategy that should be extended. So I would prefer `IDiscountStrategy`